include/structured/Dialect/Indexing/IR/IndexingOps.td

```diff
@@ -37,4 +37,33 @@ def Indexing_GatherOp : Indexing_Op<"gather", [
   }];
 }
 
+def Indexing_ConcatenateOp : Indexing_Op<"concatenate",
+    [Pure, SameOperandsAndResultElementType,
+     DeclareOpInterfaceMethods<InferTypeOpInterface>]> {
+  let summary = "Concatenate operation";
+  let description = [{
+    Concatenates a variadic number of tensors in `inputs` along `dimension`
+    dimension in the same order as the given arguments and produces a `result`
+    tensor.
+
+    Example:
+    ```mlir
+    %result = indexing.concatenate %input0, %input1, dim = 0
+        : (tensor<3x2xi64>, tensor<1x2xi64>) -> tensor<4x2xi64>
+    ```
+  }];
+
+  let arguments = (ins
+    Variadic<AnyRankedTensor>:$inputs,
+    I64Attr:$dimension
+  );
+
+  let results = (outs AnyRankedTensor);
+
+  let assemblyFormat = [{
+     custom<VariadicOperandWithAttribute>($inputs) `dim` `=` $dimension attr-dict
```

Comment:
Maybe you can change the assembly format slightly in order to avoid the custom directive. Maybe like this:

```suggestion
     `(` $inputs `)` `dim` `=` $dimension attr-dict
```

(Maybe add braces around `dim = ...`.)

Reply:
i was trying to figure out how to get it to be something like this but didn't think of delimiting with parens (not much experience yet with the parser. thanks!